src/node/rpc/member_frontend.h

```diff
@@ -447,9 +447,9 @@ namespace ccf
         cose_recent_proposals->put(key, proposal_id);
         // Only keep the most recent window_size proposals, to avoid
         // unbounded memory usage
-        if (replay_keys.size() >= window_size)
+        if (replay_keys.size() >= (window_size - 1) /* We just added one */)
```

Comment:
This was previously off by one, hidden by the fact that the test was the first to submit COSE proposals.